tripy/tripy/frontend/trace/ops/binary_elementwise.py

```diff
@@ -50,23 +50,24 @@ def __str__(self):
         return f"{self.outputs[0].name} = {op_str}"
 
     def infer_shape_output_idxs(self, inputs):
-        # permit one input to be a shape but require the output to be a shape
+        # To avoid issues, we do not permit both Shape and Tensor args and recommend explicit conversion if they do need to combined.
+        # ShapeScalar args can be broadcast up into Shapes and so do not pose an issue.
         from tripy.frontend.shape import Shape, ShapeScalar
         from tripy.utils import Result
 
         if any(map(lambda t: isinstance(t, Shape), inputs)):
-            # if there is a non-shape input, it must be rank 1 or 0 to avoid broadcasting
-            if not all(map(lambda t: t.rank <= 1, inputs)):
-                invalid_indices_message = ", ".join(
-                    map(
-                        lambda i: f"Index {i} (rank {inputs[i].rank})",
-                        filter(lambda i: inputs[i].rank > 1, range(len(inputs))),
-                    )
-                )
+            # Note: Most binary elementwise ops use the convert_inputs_to_tensors decorator, which also checks for this.
+            if any(map(lambda t: not isinstance(t, (Shape, ShapeScalar)), inputs)):
```

Comment:
Why does this need to be checked here? Wouldn't the convert decorators already have checked it?

Reply:
`min` and `max` don't use the decorator. It would otherwise be entirely redundant, yes.